tests/test_example_token.cdc

```diff
@@ -0,0 +1,157 @@
+
+    Test.expect(result, Test.beSucceeded())
+
+    var typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.TokensMinted")!
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
+
+    typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.MinterCreated")!
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
+
+    typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.TokensDeposited")!
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
+}
+
+pub fun testTransferTokens() {
+    let code = Test.readFile("../transactions/transfer_tokens.cdc")
+    let tx = Test.Transaction(
+        code: code,
+        authorizers: [recipient.address],
+        signers: [recipient],
+        arguments: [50.0, admin.address]
+    )
+
+    let result = blockchain.executeTransaction(tx)
+
+    Test.expect(result, Test.beSucceeded())
+
+    var typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.TokensDeposited")!
+    Test.assertEqual(2, blockchain.eventsOfType(typ).length)
+
+    typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.TokensWithdrawn")!
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
+}
+
+pub fun testBurnTokens() {
+    let code = Test.readFile("../transactions/burn_tokens.cdc")
+    let tx = Test.Transaction(
+        code: code,
+        authorizers: [admin.address],
+        signers: [admin],
+        arguments: [5.0]
+    )
+
+    let result = blockchain.executeTransaction(tx)
+
+    Test.expect(result, Test.beSucceeded())
+
+    var typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.BurnerCreated")!
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
+
+    typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.TokensBurned")!
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
+}
+
+pub fun testVaultTypes() {
+    let code = Test.readFile("./scripts/get_views.cdc")
+    let result = blockchain.executeScript(code, [recipient.address])
+    let viewIDs = (result.returnValue as! [String]?)!
+
+    Test.expect(result, Test.beSucceeded())
+
+    let expected = [
+        "A.01cf0e2f2f715450.FungibleTokenMetadataViews.FTView",
+        "A.01cf0e2f2f715450.FungibleTokenMetadataViews.FTDisplay",
+        "A.01cf0e2f2f715450.FungibleTokenMetadataViews.FTVaultData"
+    ]
+    Test.assertEqual(expected, viewIDs)
+}
+
+pub fun testGetVaultDisplay() {
+    let code = Test.readFile("./scripts/get_vault_display.cdc")
+    let result = blockchain.executeScript(code, [recipient.address])
+
+    Test.expect(result, Test.beSucceeded())
+}
+
+pub fun testGetVaultData() {
+    let code = Test.readFile("./scripts/get_vault_data.cdc")
+    let result = blockchain.executeScript(code, [recipient.address])
+
+    Test.expect(result, Test.beSucceeded())
+}
+
+pub fun testGetTokenMetadata() {
+    let code = Test.readFile("./scripts/get_token_metadata.cdc")
+    let result = blockchain.executeScript(code, [recipient.address])
+
+    Test.expect(result, Test.beSucceeded())
+}
```

Comment:
It would be great to verify the return values here too to make sure they are correct

Reply:
Makes sense :+1: 
I have added assertions in the test scripts: https://github.com/onflow/flow-ft/pull/137/commits/d7a27dc816bf01cad2082ca4aeaec2b781e9d969#diff-5c07d50f7e7ec4083155d1121b5006621428e4ebfe2a193475ec372126a6c631R21-R26